### ISSUE TITLE
[Bug] Non Deterministic Output

### ISSUE THREAD

**Issue by huemin-art** [nunchaku-tech/nunchaku#546]
### Checklist

- [x] 1. I have searched for related issues and FAQs (https://nunchaku.tech/docs/nunchaku/faq/faq.html) but was unable to find a solution.
- [x] 2. The issue persists in the latest version.
- [x] 3. Please note that without environment information and a minimal reproducible example, it will be difficult for us to reproduce and address the issue, which may delay our response.
- [x] 4. If your report is a question rather than a bug, please submit it as a discussion at https://github.com/mit-han-lab/nunchaku/discussions/new/choose. Otherwise, this issue will be closed.
- [x] 5. If this is related to ComfyUI, please report it at https://github.com/mit-han-lab/ComfyUI-nunchaku/issues.
- [x] 6. I will do my best to describe the issue in English.

### Describe the Bug

Outputs appear to be non-deterministic.

I am running examples/flux.1-schnell.py 

With fixed seed getting slightly different results... this is certainly a bug and I suspect comes from the custom fp16 attention implementation...

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/d5a61b3b-c26b-47c6-ba0a-51104831911c" />

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/b019fb98-12b5-404b-b9e4-006b25fb7bff" />

### Environment

Ubuntu 25.05
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 575.64.03              Driver Version: 575.64.03      CUDA Version: 12.9     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 5090        Off |   00000000:01:00.0 Off |                  N/A |
| 30%   36C    P8             12W /  575W |     141MiB /  32607MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            3720      G   /usr/bin/gnome-shell                     94MiB |
|    0   N/A  N/A            3836      G   /usr/bin/Xwayland                         9MiB |
+-----------------------------------------------------------------------------------------+

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Tue_May_27_02:21:03_PDT_2025
Cuda compilation tools, release 12.9, V12.9.86
Build cuda_12.9.r12.9/compiler.36037853_0

### Reproduction Steps

```bash
python examples/flux.1-schnell.py
```

```py
import torch
from diffusers import FluxPipeline

from nunchaku import NunchakuFluxTransformer2dModel
from nunchaku.utils import get_precision

precision = get_precision()  # auto-detect your precision is 'int4' or 'fp4' based on your GPU
transformer = NunchakuFluxTransformer2dModel.from_pretrained(
    f"mit-han-lab/nunchaku-flux.1-schnell/svdq-{precision}_r32-flux.1-schnell.safetensors"
)
pipeline = FluxPipeline.from_pretrained(
    "black-forest-labs/FLUX.1-schnell", transformer=transformer, torch_dtype=torch.bfloat16
).to("cuda")
image = pipeline(
    "A cat holding a sign that says hello world", width=1024, height=1024, num_inference_steps=4, guidance_scale=0, generator=torch.Generator(device="cuda").manual_seed(42)
).images[0]
image.save(f"flux.1-schnell-{precision}.png")
```

**Comment by lmxyy** [comment]
See https://nunchaku.tech/docs/nunchaku/faq/usage.html#why-do-the-same-seeds-produce-slightly-different-images-with-nunchaku
